File: C-Sharp/Source/Virtusales.Biblio.API.BiblioService.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Virtusales.Biblio.API
{
	public class BiblioService : BiblioAPIBase
	{
		private string ApplicationName;
		private Session Session;
		private string URL;

		public BiblioService(string applicationName, string url)
		{
			this.ApplicationName = applicationName;
			this.URL = url;
		}

		public bool Login(string username, string password, ref string errorMessage = "")
		{
			Session = new Session(ApplicationName, URL, username, password);
			if (!Session.Login(errorMessage)) {
				return false;
			}
			return true;
		}

		public object CreateRequestBasic(string page, string executionPath = "", FormattingOption responseFormat = FormattingOption.JSON)
		{
			dynamic Request = new RequestBasic(ApplicationName, URL, page, executionPath: executionPath, responseFormat: responseFormat);
			Request.Session = Session;
			return Request;
		}

		public RequestObject<TRequest, TResponse> CreateRequestObject<TRequest, TResponse>(string page, string executionPath = "")
		{
			RequestObject<TRequest, TResponse> Request = new RequestObject<TRequest, TResponse>(ApplicationName, URL, page, executionPath: executionPath);
			Request.Session = Session;
			return Request;
		}
	}
}

File: VB.NET/Source/Virtusales.Biblio.API.BiblioService.vb
Namespace Virtusales.Biblio.API
    Public Class [BiblioService] : Inherits BiblioAPIBase
        Private ApplicationName As String
        Private Session As Session
        Private URL As String
        
        Sub New(applicationName As String, url As String)
            Me.ApplicationName = applicationName
            Me.URL = url
        End Sub
        
        Public Function Login(username As String, password As String, ByRef Optional errorMessage As String = "") As Boolean
            Session = New Session(ApplicationName, URL, username, password)
            If Not Session.Login(ErrorMessage) Then
                Return False
            End If
            Return True
        End Function
        
        Public Function CreateRequestBasic(page As String, Optional executionPath As String = "", Optional responseFormat As FormattingOption = FormattingOption.[JSON])
            Dim Request = New RequestBasic(ApplicationName, URL, page, executionPath := executionPath, responseFormat := responseFormat)
            Request.Session = Session
            Return Request
        End Function
        
        Public Function CreateRequestObject(Of TRequest, TResponse)(page As String, Optional executionPath As String = "") As RequestObject(Of TRequest, TResponse)
            Dim Request As New RequestObject(Of TRequest, TResponse)(ApplicationName, URL, page, executionPath := executionPath)
            Request.Session = Session
            Return Request
        End Function
    End Class
End Namespace